### PR TITLE
Dataset file explorer to link to file contents (not file lineage)

### LINKED ISSUE #1270: Clicking on a file in the dataset view should show the preview not the lineage
Clicking on a file in the dataset view:

image

Takes me to the lineage view:

image

Instead I want to see this:

image

### PR DESCRIPTION
**Pull request by ciyer**
Fix #1270

/deploy renku-core=develop renku=2315ui-project-status #persist #notest #cypress

### CODE REVIEW

**Comment by ciyer** [comment]
# Testing

Navigate to a dataset inside a project and click on the files. That should take you to the _contents_ view (not the _lineage_ view).

File: client/src/components/FileExplorer.tsx [hunk 1]
```diff
@@ -100,12 +133,67 @@ function getFilesTree(files, foldersOpenOnLoad) {
   return treeObj;
 }
 
-class TreeNode extends Component {
-  constructor(props) {
+type FileDisplayProps = {
+  eltClassName: string;
```
Comment (leafty) [inline client/src/components/FileExplorer.tsx:137]:
```suggestion
  className: string;
```

File: client/src/components/FileExplorer.tsx [hunk 2]
```diff
@@ -100,12 +133,67 @@ function getFilesTree(files, foldersOpenOnLoad) {
...
+  eltClassName,
+  icon,
+  insideProject,
+  linkUrl,
+  node,
+}: FileDisplayProps) {
+  if (insideProject && linkUrl && node.jsonObj)
+    return (
+      <div
+        className={`fs-element ${eltClassName}`}
```
Comment (leafty) [inline client/src/components/FileExplorer.tsx:153]:
```suggestion
        className={cx("fs-element", className)}
```

and add

```ts
import cx from 'classnames';
```

File: client/src/components/FileExplorer.tsx [hunk 3]
```diff
@@ -235,30 +314,34 @@ function FilesTreeView(props) {
...
- * @param {*} props.files - This is a list of files with atLocation containing the file path (this is optional)
- * @param {*} props.filesTree - This is the already built fileTree (optional)
- * @param {*} props.foldersOpenOnLoad - Number of folders that should appear open already when displaying the tree
- * @param {*} props.lineageUrl - Should be replaced for URL, this is the link for the file (when clicked) (optional)
- * @param {*} props.insideProject - Boolean to be set true if the display is inside a project
- */
-function FileExplorer(props) {
-  const [filesTree, setFilesTree] = useState(undefined);
+type FileExplorerProps = {
+  files?: JsonObj[]; // This is a list of files with atLocation containing the file path (this is optional)
```
Comment (leafty) [inline client/src/components/FileExplorer.tsx:318]: These should be jsDoc comments, e.g. they will be displayed by vscode when hovering the prop.
```suggestion
  /** List of files with atLocation containing the file path (this is optional) */
  files?: JsonObj[];
```

**Review by leafty — APPROVED** [review]
LGTM it seems to work as intended

Some minor comments below.

File: client/src/components/FileExplorer.tsx [hunk 4]
```diff
@@ -151,57 +238,48 @@ class TreeNode extends Component {
         })
       : null;
 
-    let elementToRender;
-    const eltClassName = order + " " + hidden;
+    const className = order + " " + hidden;
```
Comment (leafty) [inline client/src/components/FileExplorer.tsx:241]: I noticed this should also use `cx`:
```ts
const order = this.props.node.children.length
  ? "order-second"
  : "order-third";

const hidden = this.props.node.name.startsWith(".") && "hidden-folder";

const className = cx(order, hidden);
```

The `cx` function will handle falsy values properly (as opposed to having `undefined` being one of the CSS class names which is something we have in the UI and should fix)

**Pull request merged**